prow/bugzilla/client.go

```diff
@@ -75,25 +79,77 @@ type Client interface {
 	RemovePullRequestAsExternalBug(id int, org, repo string, num int) (bool, error)
 	// GetAllClones returns all the clones of the bug including itself
 	// Differs from GetClones as GetClones only gets the child clones which are one level lower
-	GetAllClones(bug *Bug) ([]*Bug, error)
+	// GetAllClones will also fetch bugDetails and the root bug details in addition to clones
+	// since we want to parallelize all the HTTP calls
+	GetAllClones(bugID int) (clones []*Bug, bug *Bug, root *Bug, err error)
 	// GetRootForClone returns the original bug.
 	GetRootForClone(bug *Bug) (*Bug, error)
+	// GetCachedClones will fetch the clones from the clones cache
+	GetCachedClones(bugID int) ([]int, bool)
+	// SetCachedClones will push the cloneIDs associated with the bug to the cache
+	SetCachedClones(bugID int, clonesList []int)
 }
 
 func NewClient(getAPIKey func() []byte, endpoint string) Client {
 	return &client{
-		logger:    logrus.WithField("client", "bugzilla"),
-		client:    &http.Client{},
-		endpoint:  endpoint,
-		getAPIKey: getAPIKey,
+		logger:      logrus.WithField("client", "bugzilla"),
+		client:      &http.Client{},
+		endpoint:    endpoint,
+		getAPIKey:   getAPIKey,
+		clonesCache: clonesCacheWithMutex{cache: map[int][]int{}},
 	}
 }
 
+type clonesCacheWithMutex struct {
```

Comment:
The `WithMutex` in the name is a  superfluous implementation detail, the Bugzilla client is also not named `BugZillaClientWithHTTPClient`